cmake/deps/libbacktrace.cmake

```diff
@@ -1,3 +1,7 @@
-target_compile_definitions(clio_options INTERFACE BOOST_STACKTRACE_LINK)
-target_compile_definitions(clio_options INTERFACE BOOST_STACKTRACE_USE_BACKTRACE)
-find_package(libbacktrace REQUIRED CONFIG)
+if (Boost_STACKTRACE_FOUND)
+  target_compile_definitions(clio_options INTERFACE BOOST_STACKTRACE_LINK)
+  target_compile_definitions(clio_options INTERFACE BOOST_STACKTRACE_USE_BACKTRACE)
+  find_package(libbacktrace REQUIRED CONFIG)
+else ()
+  target_compile_definitions(clio_options INTERFACE CLIO_WITHOUT_STACKTRACE)
```

Comment:
Some sanitizers (TSAN and ASAN for sure) can't be used with libbacktrace because they have their own backtracing capabilities and there are conflicts. In any case, this makes sure Clio code knows that backtrace is not available.